pandas/tseries/offsets.py

```diff
@@ -221,6 +223,39 @@ def apply(self, other):
         else:
             return other + timedelta(self.n)
 
+    def apply_index(self, i):
+        relativedelta_fast = set(['years', 'months', 'weeks',
```

Comment:
It's 4 lines down - if a keyword was passed to `DateOffset` that's valid in `relativedelta` but I can't handle (e.g. singular `day` argument) this is used to fallback.